Observer/ProductSaveAfter.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Exception;
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Klaviyo\Reclaim\Model\ProductsFactory;
+
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+
+class ProductSaveAfter implements ObserverInterface
+{
+    /**
+     * Klaviyo scope setting helper
+     * @var ScopeSetting $klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * Klaviyo product factory
+     * @var klProductFactory
+     */
+    protected $_klProductFactory;
+
+    /**
+     * Magento stock registry api interface
+     * @var $stockRegistry
+     */
+    protected $_stockRegistry;
+
+    /**
+     * @param ScopeSetting $klaviyoScopeSetting
+     * @param CategoryFactory $categoryFactory
+     * @param ProductsFactory $klProductFactory
+     * @param StockRegistryInterface $stockRegistry
+     */
+    public function __construct(
+        ScopeSetting $klaviyoScopeSetting,
+        ProductsFactory $klProductFactory,
+        StockRegistryInterface $stockRegistry
+    ) {
+        $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
+        $this->_klProductFactory = $klProductFactory;
+        $this->_stockRegistry = $stockRegistry;
+    }
+
+    /**
+     * customer register event handler
+     *
+     * @param Observer $observer
+     * @return void
+     * @throws Exception
+     */
+    public function execute(Observer $observer)
+    {
+        $product = $observer->getEvent()->getProduct();
+        $storeIds = $product->getStoreIds();
+        $storeIdKlaviyoMap = $this->_klaviyoScopeSetting->getStoreIdKlaviyoAccountSetMap($storeIds);
+
+        foreach ($storeIdKlaviyoMap as $klaviyoId => $storeIds) {
+            if (empty($storeIds)) {continue;}
+
+            if ($this->_klaviyoScopeSetting->getWebhookSecret() && $this->_klaviyoScopeSetting->getProductSaveWebhookSetting($storeIds[0])) {
+              $normalizedProduct = $this->normalizeProduct($product);
+              $data = [
+                'status'=>'NEW',
+                'topic'=>'product/save',
+                'klaviyo_id'=>$klaviyoId,
+                'payload'=>json_encode($normalizedProduct)
+              ];
+              $klProduct = $this->_klProductFactory->create();
+              $klProduct->setData($data);
+              $klProduct->save();
+            }
+        }
+    }
+
+    private function normalizeProduct($product=null)
+    {
+      if ($product == null) {return;}
+
+      $product_id = $product->getId();
+
+      $product_info = array(
+        'store_ids' => $product->getStoreIds(),
```

Comment:
Are all array_keys supposed to start lower case/upper case? Could we standardize across the payload?

Reply:
I only use uppercase letters on some of the properties because Klaviyo wants them that way so this avoids normalizing the payload in the app. otherwise I user lowercase where I can. Let me know if that's alright or not!

Reply:
Is there any restriction on making other start upper case too? Just looking for standardize keys. it's a nitpick so if it is too much work to accommodate this, we can skip it